### ISSUE TITLE
Question about WSS and SSH

### ISSUE THREAD

**Issue by killwing** [jpillora/chisel#325]
"Chisel is a fast TCP/UDP tunnel, transported over HTTP, secured via SSH."
If the connection is setup by secured websocket (wss), is SSH still needed?

**Comment by techiek7** [comment]
Can anyone please answer ?

**Comment by redneck-f25** [comment]
Yes, SSH is essentially needed. SSH is more than authentication and encryption. It also includes reliable multiplexing for multiple channels through one tunnel (tcp-connection). Chisel uses this part of the well-documented protocol and well-testet library to provide the Remotes. If transportet over TLS (HTTPS) the Encryption-Part of SSH is not required, but included per specification. You might try to use a null-cipher to reduce the load.

- [How can I disable encryption on openssh?](https://serverfault.com/questions/116875/how-can-i-disable-encryption-on-openssh)
- https://pkg.go.dev/github.com/keltia/cipher/null
- https://github.com/keltia/cipher/blob/master/null/cipher.go
- https://pkg.go.dev/golang.org/x/crypto/ssh#Config
